### PR TITLE
Cask Hydrator: Re-organize plugins: add analytics

### PR DESCRIPTION
**Pull request by awholegunch**
Running [Quick Build 2](http://builds.cask.co/browse/CDAP-DQB94-2)

[Analytic Plugins](http://builds.cask.co/artifact/CDAP-DQB94/shared/build-2/Docs-HTML/3.5.0/en/hydrator-manual/plugins/analytics/index.html)

### CODE REVIEW

File: cdap-docs/hydrator-manual/source/_includes/plugins/analytics/index.rst [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+These plugins work with:
+
+  - *Data Pipeline - Batch* (``cdap-data-pipeline`` artifact)
+  - *Data Pipeline - Realtime* (``cdap-data-streams`` artifact)
+
+.. toctree::
+    :maxdepth: 1
+    :glob:
+
+    [!D][!P][!B]*
```
Comment (albertshau) [inline cdap-docs/hydrator-manual/source/_includes/plugins/analytics/index.rst:22]: if I understand correctly, this includes all the transforms? In which case it is wrong.

analytics for batch = batchaggregator, batchjoiner, sparkcompute, sparksink
analytics for realtime = batchaggregator, batchjoiner, sparkcompute

Using the DPB prefix is a little confusing. Does something like this work?

```
  *-batchaggregator.md
  *-batchjoiner.md
  *-sparkcompute.md
```

File: cdap-docs/hydrator-manual/source/_includes/plugins/analytics/index.rst [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    
+These plugins work with:
+
+  - *Data Pipeline - Batch* (``cdap-data-pipeline`` artifact)
+
+.. toctree::
+    :maxdepth: 1
+    :glob:
+
+    DPB*
```
Comment (albertshau) [inline cdap-docs/hydrator-manual/source/_includes/plugins/analytics/index.rst]: only type that batch supports that realtime doesn't is sparksink

File: cdap-docs/hydrator-manual/source/_includes/plugins/analytics/index.rst (hunk 1 above)
Comment (albertshau) [inline cdap-docs/hydrator-manual/source/_includes/plugins/analytics/index.rst:22]: ok I understand now, DPB is only put in front of the spark compute plugin. It should be the spark sink plugin though.

File: cdap-docs/hydrator-manual/source/_includes/plugins/analytics/index.rst [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+.. - `NaiveBayesClassifier <naivebayesclassifier-sparkcompute.html>`__
```
Comment (albertshau) [inline cdap-docs/hydrator-manual/source/_includes/plugins/analytics/index.rst]: is this a comment? or is it an actual link that will be placed here?

Comment (awholegunch) [reply cdap-docs/hydrator-manual/source/_includes/plugins/analytics/index.rst]: Comment; can be removed. Fixed.

File: cdap-docs/hydrator-manual/source/_includes/plugins/analytics/index.rst (hunk 1 above)
Comment (awholegunch) [reply cdap-docs/hydrator-manual/source/_includes/plugins/analytics/index.rst:22]: Globbing is limited, and doesn't support patterns as you suggest. Fixed the plugin prefixed for the `DPB` (Data Pipeline Batch).

File: cdap-docs/hydrator-manual/build.sh [hunk 4]
```diff
@@ -283,8 +283,8 @@ function download_includes() {
   download_md_file mongodb-plugins MongoDB-realtimesink.md
   download_md_file spark-plugins Kafka-streamingsource.md
   download_md_file spark-plugins Twitter-streamingsource.md
-  download_md_file spark-plugins NaiveBayesClassifier-sparkcompute.md '' "transform" "DPB-naivebayesclassifier-sparkcompute.md" # Currently only for batch
-  download_md_file spark-plugins NaiveBayesTrainer-sparksink.md       '' "sink" # Currently only for batch
+  download_md_file spark-plugins NaiveBayesClassifier-sparkcompute.md '' "analytic" # Currently only for batch
```
Comment (albertshau) [inline cdap-docs/hydrator-manual/build.sh]: the comment 'Currently only for batch' is no longer true

**Comment by albertshau** [comment]
one very minor comment. lgtm, fix then merge.

**Pull request merged**